examples/json.rs

```diff
@@ -0,0 +1,29 @@
+use async_std::io;
+use async_std::task;
+use serde::{Deserialize, Serialize};
+
+#[derive(Deserialize, Serialize)]
+struct Cat {
+    name: String,
+}
+
+fn main() -> io::Result<()> {
+    task::block_on(async {
+        let mut app = tide::new();
+
+        app.at("/submit").post(|mut req: tide::Request<()>| {
+            async move {
+                let cat: Cat = req.body_json().await.unwrap();
```

Comment:
You changed the tide error handling but is it always possible to `.client_error()?;` with it ?

Reply:
I'm unsure. Directly using ? definitely isn't. If you could try it out I'd love to hear about it!

Reply:
I just tried and it doesn't seems possible, at least with just an `?` at the end of a line :)